target/linux/realtek/dts/rtl8380_netgear_gs108t-v3.dts

```diff
@@ -0,0 +1,202 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+// Memory
+// ------
+// Spec: 256 MB DDR3-1600
+// https://www.winbond.com/resource-files/W631GG8MB_A03.pdf
+//
+// W631GG8MB-12 DDR3-1600 (11-11-11)
+// The W631GG8MB is a 1G bits DDR3 SDRAM, organized as 16,777,216 words x 8 banks x 8 bits.
+// 1KB page size
+//
+// Flash
+// -----
+// Spec: 32 MB 3v SPI
+// https://www.winbond.com/resource-files/w25q256jv%20spi%20revb%2009202016.pdf
+//
+// PACKAGE TYPE: F - SOIC-16 300mil
+// DENSITY: 256M-bit
+// PRODUCT NUMBER: W25Q256JVFIQ
+// TOP SIDE MARKING: 25Q256JVFQ
+
+/dts-v1/;
+
+#include "rtl838x.dtsi"
+
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	compatible = "netgear,gs108t-v3", "realtek,rtl838x-soc";
+	model = "Netgear GS108T";
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		mode {
+			label = "reset";
+			gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+};
+
+&gpio0 {
+	indirect-access-bus-id = <0>;
+};
+
+&spi0 {
+	status = "okay";
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+			partition@0 {
+				label = "u-boot";
+				reg = <0x0000000 0x0e0000>;
+				read-only;
+			};
+			partition@e0000 {
+				label = "u-boot-env";
+				reg = <0x00e0000 0x010000>;
+			};
+			partition@f0000 {
+				label = "sysinfo";
+				reg = <0x00f0000 0x010000>;
+				read-only;
+			};
+			partition@100000{
+				label = "jffs2_cfg";
+				reg = <0x0100000 0x100000>;
+				read-only;
+			};
+			partition@200000{
+				label = "jffs2_log";
+				reg = <0x0200000 0x100000>;
+			};
+			partition@300000{
+				label = "firmware";
+				compatible = "netgear,uimage";
+				reg = <0x0300000 0x1d00000>;
+			};
+		};
+	};
+};
+
+&ethernet0 {
+	mdio: mdio-bus {
+		compatible = "realtek,rtl838x-mdio";
+		regmap = <&ethernet0>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		/* Internal phy */
+		phy8: ethernet-phy@8 {
```

Comment:
rtl838x.dtsi has a few macros you can use to make this more concise.

Reply:
Done, for this and the switch ports as well.